app/search/routes.py

```diff
@@ -137,18 +138,43 @@ def dataset_search():
     queryAll = bool(request.args.get('elements') == 'all')
 
     if(not queryAll):
-        delta = int(request.args.get('max_per_page', 10)) * (int(request.args.get('page', 1)) - 1 )
+
+        if request.args.get('modalities'):
+            filterModalities = request.args.get('modalities').split(",")
+            elements = list(filter(lambda e: e['modalities'] is not None, elements))
+            elements = list(filter(lambda e: all(item in e['modalities'].lower() for item in filterModalities), elements))
+            print(elements)
+        if request.args.get('formats'):
+            filterFormats = request.args.get('formats')
+            elements = list(filter(lambda e: e['format'] is not None, elements))
+            elements = list(filter(lambda e: all(item in e['format'].lower() for item in filterFormats), elements))
+
+        delta = int(request.args.get('max_per_page', 10)) * \
+                    (int(request.args.get('page', 1)) - 1)
         cursor = max(min(int(request.args.get('cursor') or 0), 0), 0) + delta
         limit = max(min(int(request.args.get('limit') or 10), 10), 0)
         sort_key = request.args.get('sortKey') or "conpStatus"
         paginated = elements
-        paginated.sort(key=lambda o: (o[sort_key] is None, o[sort_key]))
-        paginated = paginated[(cursor):(cursor + limit)]
+
+        if(sort_key == "conpStatus"):
+            order = {'conp': 0, 'canadian': 1, 'external': 2}
+            paginated.sort(key=lambda o: order[o[sort_key].lower()])
+
+        elif(sort_key == "title"):
+            paginated.sort(key=lambda o: o[sort_key].lower())
+
+        elif(sort_key == "size"):
```

Comment:
This is not working correctly. It should support missing values (as in the final else) and 1 GB shows as smaller than 194 MB.
Sorting by size could be done in subsequent PR since it is not trivial (unit convertion).

Reply:
I was going to look at sizes as a separate PR, but it might make more sense to do it here. Let me take a look.